src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -1012,7 +977,7 @@ public void PersistOnFlush(object obj)
 
 		public override string BestGuessEntityName(object entity)
 		{
-			using (new SessionIdLoggingContext(SessionId))
+			using (BeginProcess())
```

Comment:
Does access the `persistenceContext`, thus my choice of switching it to `BeginProcess`. Now what it does likely can work on closed session, so I can revert it to a `BeginContext` if asked to.